CBot/src/CBot/CBotVar/CBotVar.h

```diff
@@ -135,31 +142,20 @@ class CBotVar : public CBotLinkedList<CBotVar>
 
     ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-    /**
-     * \brief Associates an unique identifier to class instance
-     *
-     * Used only by classes
-     *
-     * \param UniqId New unique identifier
-     * \see SetUniqNum() for another identifier, used for all variable types
-     */
-    virtual void SetIdent(long UniqId);
-
-    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
     //! \name User pointer
     //@{
 
     /**
      * \brief Set a custom pointer associated with this variable
-     * \param pUser custom pointer to set
+     * \param user custom pointer to set
      */
-    void SetUserPtr(void* pUser);
+    virtual void SetUserPointer(std::unique_ptr<CBotUserPointer> user);
 
     /**
      * \brief Returns the custom pointer associated with this variable
-     * \return A pointer set with SetUserPtr()
+     * \return A pointer set with SetUserPointer()
      */
-    void* GetUserPtr();
+    virtual const std::unique_ptr<CBotUserPointer>& GetUserPointer();
```

Comment:
```suggestion
    virtual std::optional<void *> GetUserPointer();
```

See my comment on CBotVarClass.h